p2p.go

```diff
@@ -380,9 +402,36 @@ func NewHost(ctx context.Context, options ...Option) (*Host, error) {
 }
 
 // JoinOverlay triggers the host to join the DHT overlay
-func (h *Host) JoinOverlay(ctx context.Context) {
+func (h *Host) JoinOverlay(ctx context.Context) error {
 	routingDiscovery := discovery.NewRoutingDiscovery(h.kad)
-	discovery.Advertise(ctx, routingDiscovery, h.kadKey.String())
+	h.peerManager = newPeerManager(h.host, routingDiscovery, h.cfg.GroupID, 100)
+	return h.peerManager.Advertise(ctx)
+}
+
+// ConnectPeers connect host to the peers with the same groupID
+func (h *Host) ConnectPeers(ctx context.Context) error {
+	if h.peerManager == nil {
+		return errNoPeerManager
+	}
+	if err := h.peerManager.ConnectPeers(ctx); err != nil {
+		return err
+	}
+	// setup a goroutine by caller?
+	// TODO: stop goroutine when p2p stop()
```

Comment:
yeah, i think this go-routine can be done in core's p2p using a `RecurringTask`?